dipy/segment/tests/test_clustering.py

```diff
@@ -0,0 +1,575 @@
+    assert_array_equal(list(clusters[subset][1]), clusters2_indices)
+
+
+def test_cluster_map_centroid_attributes_and_constructor():
+    clusters = ClusterMapCentroid()
+    assert_array_equal(clusters.centroids, [])
+    assert_raises(AttributeError, setattr, clusters, 'centroids', [])
+
+
+def test_cluster_map_centroid_add_cluster():
+    clusters = ClusterMapCentroid()
+
+    centroids = []
+    for i in range(3):
+        cluster = ClusterCentroid(centroid=np.zeros_like(features))
+
+        centroids.append(np.zeros_like(features))
+        for id_data in range(2*i):
+            centroids[-1] = (centroids[-1]*id_data + (id_data+1)*features) / (id_data+1)
+            cluster.assign(id_data, (id_data+1)*features)
+            cluster.update()
+
+        clusters.add_cluster(cluster)
+        assert_array_equal(cluster.centroid, centroids[-1])
+        assert_equal(type(cluster), ClusterCentroid)
+        assert_equal(cluster, clusters[-1])
+
+    assert_equal(type(clusters.centroids), list)
+    assert_array_equal(list(itertools.chain(*clusters.centroids)), list(itertools.chain(*centroids)))
+
+    # Check adding features of different sizes (shorter and longer)
+    features_shape_short = (1, features_shape[1]-3)
+    features_too_short = np.ones(features_shape_short, dtype=dtype)
+    assert_raises(ValueError, cluster.assign, 123, features_too_short)
+
+    features_shape_long = (1, features_shape[1]+3)
+    features_too_long = np.ones(features_shape_long, dtype=dtype)
+    assert_raises(ValueError, cluster.assign, 123, features_too_long)
+
+
+def test_cluster_map_centroid_remove_cluster():
+    clusters = ClusterMapCentroid()
+
+    centroid1 = np.random.rand(*features_shape).astype(dtype)
+    cluster1 = ClusterCentroid(centroid1, indices=[1])
+    clusters.add_cluster(cluster1)
+
+    centroid2 = np.random.rand(*features_shape).astype(dtype)
+    cluster2 = ClusterCentroid(centroid2, indices=[1, 2])
+    clusters.add_cluster(cluster2)
+
+    centroid3 = np.random.rand(*features_shape).astype(dtype)
+    cluster3 = ClusterCentroid(centroid3, indices=[1, 2, 3])
+    clusters.add_cluster(cluster3)
+
+    assert_equal(len(clusters), 3)
+
+    clusters.remove_cluster(cluster2)
+    assert_equal(len(clusters), 2)
+    assert_array_equal(list(itertools.chain(*clusters)), list(itertools.chain(*[cluster1, cluster3])))
+    assert_array_equal(clusters.centroids, np.array([centroid1, centroid3]))
+    assert_equal(clusters[0], cluster1)
+    assert_equal(clusters[1], cluster3)
+
+    clusters.remove_cluster(cluster3)
+    assert_equal(len(clusters), 1)
+    assert_array_equal(list(itertools.chain(*clusters)), list(cluster1))
+    assert_array_equal(clusters.centroids, np.array([centroid1]))
+    assert_equal(clusters[0], cluster1)
+
+    clusters.remove_cluster(cluster1)
+    assert_equal(len(clusters), 0)
+    assert_array_equal(list(itertools.chain(*clusters)), [])
+    assert_array_equal(clusters.centroids, [])
+
```

Comment:
No, it was not possible to add nor remove multiple clusters at once. After the next commit, it will though :stuck_out_tongue_winking_eye:.